### PR TITLE
build: fix v_reduce_sum4 compilation (requires SSE3)

### PR DESCRIPTION
**Pull request by alalek**
Related #8907

### CODE REVIEW

**Comment by alalek** [comment]
:+1:

File: modules/core/include/opencv2/core/hal/intrin_sse.hpp [hunk 1]
```diff
@@ -1129,9 +1129,13 @@ OPENCV_HAL_IMPL_SSE_REDUCE_OP_4_SUM(v_float32x4, float, __m128, ps, _mm_castps_s
 inline v_float32x4 v_reduce_sum4(const v_float32x4& a, const v_float32x4& b,
                                  const v_float32x4& c, const v_float32x4& d)
 {
+#if CV_SSE3
     __m128 ab = _mm_hadd_ps(a.val, b.val);
     __m128 cd = _mm_hadd_ps(c.val, d.val);
     return v_float32x4(_mm_hadd_ps(ab, cd));
+#else
+    return v_float32x4(v_reduce_sum(a), v_reduce_sum(b), v_reduce_sum(c), v_reduce_sum(d));
```
Comment (vpisarev) [inline modules/core/include/opencv2/core/hal/intrin_sse.hpp]: let's make it more efficient:

<pre>
__m128 ac = _mm_add_ps(_mm_unpacklo_ps(a.val, c.val), _mm_unpackhi_ps(a.val, c.val));
__m128 bd = _mm_add_ps(_mm_unpacklo_ps(b.val, d.val), _mm_unpackhi_ps(b.val, d.val));
return v_float32x4(_mm_add_ps(_mm_unpacklo_ps(ac, bd), _mm_unpackhi_ps(ac, bd)));
</pre>

Comment (alalek) [inline modules/core/include/opencv2/core/hal/intrin_sse.hpp]: Thanks. Updated.

**Comment by vpisarev** [comment]
you are the assignee, but here is my :+1:

File: modules/core/test/test_intrin.cpp [hunk 2]
```diff
@@ -741,6 +741,26 @@ template<typename R> struct TheTest
...
+    {
+        R a(0.1f, 0.02f, 0.003f, 0.0004f);
+        R b(1, 20, 300, 4000);
+        R c(10, 2, 0.3f, 0.04f);
+        R d(1, 2, 3, 4);
+
+        R sum = v_reduce_sum4(a, b, c, d);
+
+        Data<R> res = sum;
+        for (int i = 0; i < R::nlanes; ++i)
```
Comment (mshabunin) [inline modules/core/test/test_intrin.cpp]: I think this loop is not needed here.

Comment (alalek) [inline modules/core/test/test_intrin.cpp]: Removed.

**Pull request merged**